satpy/readers/hdfeos_base.py

```diff
@@ -0,0 +1,240 @@
+        except KeyError as e:
+            logger.debug("Resolution not found in metadata: {}".format(e))
+            return None
+        resolution_regex = re.compile(r'(?P<resolution>\d+)(km|KM)')
+        resolution_match = resolution_regex.search(latitude_dim)
+        return int(resolution_match.group('resolution')) * 1000
+
+    @property
+    def geo_resolution(self):
+        """Resolution of the geographical data retrieved in the metada. """
+        return self.read_geo_resolution(self.metadata)
+
+    def get_dataset(self, dataset_keys, dataset_info):
+        """Get the geolocation dataset."""
+        # Name of the dataset as it appears in the HDF EOS file
+        in_file_dataset_name = dataset_info['file_key']
+        # Name of the dataset in the YAML file
+        dataset_name = dataset_keys.name
+        # Resolution asked
+        resolution = dataset_keys.resolution
+
+        data = self.load_dataset(in_file_dataset_name)
+
+        if resolution != self.geo_resolution:
+
+            # The data must be interpolated
+            interpolated_dataset = {}
+
+            def interpolate(clons, clats, csatz):
+                from geotiepoints.modisinterpolator import modis_1km_to_250m, modis_1km_to_500m, modis_5km_to_1km
+
+                interpolation_functions = {
+                    (5000, 1000): modis_5km_to_1km,
+                    (1000, 500): modis_1km_to_500m,
+                    (1000, 250): modis_1km_to_250m
+                }
+
+                try:
+                    interpolation_function = interpolation_functions[(self.geo_resolution, resolution)]
+                except KeyError:
+                    error_message = "Interpolation from {}m to {}m not implemented".format(
+                        self.geo_resolution, resolution)
+                    raise NotImplementedError(error_message)
+
+                logger.debug("Interpolating from {} to {}".format(self.geo_resolution, resolution))
+
+                return interpolation_function(clons, clats, csatz)
+
+            # Sensor zenith dataset name differs between L1b and L2 products
+            sensor_zentih = None
+            try:
+                sensor_zenith = self.load_dataset('SensorZenith')
+            except KeyError:
+                sensor_zenith = self.load_dataset('Sensor_Zenith')
+
+            if dataset_name in ['longitude', 'latitude']:
+                latitude = self.load_dataset('Longitude')
+                longitude = self.load_dataset('Latitude')
+                longitude, latitude = interpolate(
+                    longitude, latitude, sensor_zenith
+                )
+                interpolated_dataset['longitude'] = longitude
+                interpolated_dataset['latitude'] = latitude
+
+            # Warning: Are these interpolations originally correct?
+            # Does geotiepoints actually interpolate azimuth coordinates?
```

Comment:
Yes, it should be fine. The names of the variables are just misleading, and should be fixed sorry.

Reply:
I will remove those comments then.